confidant/core/secrets.py

```
@@ -0,0 +1,120 @@
+        decrypt_secret("ENC(...)") -> "my-secret"
+    """
+    try:
+        if not (cipher_text.startswith("ENC(") and cipher_text.endswith(")")):
+            raise SecretsError("Invalid secret format. Expected ENC(...).")
+
+        encrypted_part = cipher_text[4:-1]
+        fernet = Fernet(_get_master_key())
+        decrypted = fernet.decrypt(encrypted_part.encode()).decode()
+        return decrypted
+    except InvalidToken:
+        raise SecretsError("Invalid token. Decryption failed. Check your master key.")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)